### PR TITLE
xds-k8s: Link xDS test client and test server logs in Logs Explorer

### PR DESCRIPTION
**Pull request by sergiitk**
Long-awaited feature that links xDS test client and test server logs in Logs Explorer.

Test run: https://source.cloud.google.com/results/invocations/2f1fa1d2-ab15-488b-a927-3f5f13e70293/targets/grpc%2Fjava%2Fmaster%2Fbranch%2Fxds-k8s/log

Search for `GCP Logs Explorer link`:
image

### CODE REVIEW

File: tools/run_tests/xds_k8s_test_driver/framework/test_app/base_runner.py [hunk 1]
```diff
@@ -287,8 +306,36 @@ def _wait_service_neg(self, name, service_port, **kwargs):
...
+            'resource.labels.project_id': gcp_project,
+            'resource.labels.container_name': deployment_name,
+            'resource.labels.namespace_name': namespace_name,
+        })
+        request = _logs_explorer_request({
+            'query': query,
+            'timeRange': f'{time_now}/{time_end}',
+        })
+
+        log_link = ('https://pantheon.corp.google.com/logs/query;'
```
Comment (gnossen) [inline tools/run_tests/xds_k8s_test_driver/framework/test_app/base_runner.py:332]: It's a bit odd seeing `corp.google.com` in an OSS test. Is there any way that we can abstract away pantheon vs. GCP? I think `console.cloud.google.com` will redirect to Pantheon if you have an `@google.com` email.

Also, you [probably want to pull this string out to a module-level `_NAMED_CONSTANT`](https://google.github.io/styleguide/pyguide.html#254-decision).

**Review by gnossen — APPROVED** [review]
Just one nit. Thanks for the awesome feature!

File: tools/run_tests/xds_k8s_test_driver/framework/test_app/base_runner.py (hunk 1 above)
Comment (sergiitk) [reply tools/run_tests/xds_k8s_test_driver/framework/test_app/base_runner.py:332]: @gnossen This is really good point, I completely forgot we use dedicated domain.
Unfortunately, the redirect from `console.cloud.google.com` is buggy and drops the query part 😕

I'll make it an argument instead that defaults to `console.cloud.google.com`, and override it in `grpc-testing` specific configs.

Comment (sergiitk) [reply tools/run_tests/xds_k8s_test_driver/framework/test_app/base_runner.py:332]: Redirect example: https://console.cloud.google.com/logs/query;query=resource.type%3D%22k8s_container%22%0Aresource.labels.project_id%3D%22grpc-testing%22%0Aresource.labels.container_name%3D%22psm-grpc-server%22%0Aresource.labels.namespace_name%3D%22xds-k8s-security-server-20210730-1901-5t3ux%22;timeRange=2021-07-30T19:02:35.717526Z%2F2021-07-30T20:02:35.717556Z?project=grpc-testing

**Pull request merged**